### PR TITLE
Deploy Docs on Master Builds

### LINKED ISSUE #1641: Add Continuous Docs Release
Based on https://medium.com/@nthgergo/publishing-gh-pages-with-travis-ci-53a8270e87db.

@stefanv, what is the current process for pushing the latest dev documents?

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #1641.  Replaces #990.

### CODE REVIEW

File: tools/deploy_docs.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+if [[ $TRAVIS_PULL_REQUEST == false && $TRAVIS_BRANCH == "master" ]]
+then
+    echo "-- pushing docs --"
+
+    (
+    git config user.email "travis@travis-ci.com"
+    git config user.name "Travis Bot"
+
+    git clone --quiet --branch=gh-pages https://${GH_REF} doc_build
```
Comment (blink1073) [inline tools/deploy_docs.sh]: I tested this sequence of commands on my machine, but we won't know for sure if it works until someone hits the big green button.  :smile:

File: tools/deploy_docs.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    rm -r doc_build/dev
+    cp -r doc/build/html doc_build/dev
+
+    cd doc_build
+    git add dev
+    git commit -m "Deployed to GitHub Pages"
+    git push --force --quiet "https://${GHTOKEN}@${GH_REF}" master:gh-pages > /dev/null 2>&1
```
Comment (emmanuelle) [inline tools/deploy_docs.sh]: Did you also try this line (to push)? I can see on https://github.com/scikit-image/docs/tree/gh-pages that the last commit dates back to July 16?

Comment (blink1073) [inline tools/deploy_docs.sh]: No, I actually had an error while building the docs that I'll have to come back to.

Comment (blink1073) [inline tools/deploy_docs.sh]: Turns out it was a bug because we had a module named `notebook`, which is now preempted by Jupyter :smile:

**Comment by stefanv** [comment]
OK, here we go :)  Thank you very much, Steven, this has been a long time requested feature.

We should document the procedure for generating the token somewhere.

**Pull request merged**